### PR TITLE
[harfbuzz] Update to 2.6.4

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fix related issue https://github.com/microsoft/vcpkg/issues/10075

1. Update harfbuzz to latest revision 2.6.4, the latest revision contains source fix.
2. Update the patch to adjust the source changes.

All features test pass on x86-windows.

### CODE REVIEW

File: ports/harfbuzz/portfile.cmake [hunk 1]
```diff
@@ -65,6 +46,7 @@ vcpkg_configure_cmake(
         -DHB_HAVE_GLIB=${HAVE_GLIB}
         -DHB_HAVE_GRAPHITE2=${HB_HAVE_GRAPHITE2}
```
Comment (c72578) [inline ports/harfbuzz/portfile.cmake]: `HB_HAVE_GRAPHITE2` is part of the `FEATURE_OPTIONS`

Comment (PhoebeHui) [reply ports/harfbuzz/portfile.cmake]: Thanks! I forget to remove it.

**Comment by c72578** [comment]
Concerning `HB_BUILTIN_UCDN`, the ChangeLog file says the following:
```console
commit 65392b734e38668b870b1ffcbfb4b42ec289ef58
Author: Behdad Esfahbod <behdad@behdad.org>
Date:   Wed May 22 16:21:21 2019 -0400

    [ucdn] Replace UCDN with a new UCD implementation
```
So, `ucdn       HB_BUILTIN_UCDN` can be removed from `FEATURE_OPTIONS`.

**Comment by PhoebeHui** [comment]
@c72578, thanks for pointing out this! I will update it.

**Comment by PhoebeHui** [comment]
It seems that the funcutions related to UCD enabled by default. @past-due, could you help review? I noticed you add the feature to vcpkg before, related to https://github.com/microsoft/vcpkg/pull/3560

File: ports/harfbuzz/portfile.cmake [hunk 2]
```diff
@@ -26,45 +24,19 @@ if("${_contents}" MATCHES "find_library\\(GLIB_LIBRARIES")
...
-    set(BUILTIN_UCDN ON)
-endif()
-
-# Glib
-set(HAVE_GLIB OFF)
-if("glib" IN_LIST FEATURES)
-    set(HAVE_GLIB ON)
-endif()
-
-# At least one Unicode callback must be specified, or harfbuzz compilation fails
```
Comment (past-due) [inline ports/harfbuzz/portfile.cmake]: 👍

**Review by past-due — COMMENTED** [review]
The Harfbuzz port changes look good to me.

File: ports/harfbuzz/portfile.cmake (hunk 2 above)
Comment (PhoebeHui) [reply ports/harfbuzz/portfile.cmake]: Thanks for your review!

**Comment by pkgw** [comment]
The errors with the current harfbuzz rules are causing problems for my project's CI. What can I do to help move this PR forward?

**Comment by LilyWangL** [comment]
Duplicate with #11416

**Pull request closed without merging**